sdk/go/common/util/executable/windows.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2025, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build windows
+
+package executable
+
+import (
+	"errors"
+	"fmt"
+	"syscall"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+)
+
+func IsExecutable(path string) (bool, error) {
+	// Use GetBinaryTypeW to determine if the file is an executable.
+	// https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-getbinarytypew
+	pathp, err := windows.UTF16PtrFromString(path)
+	if err != nil {
+		return false, fmt.Errorf("unable to convert path to UTF16: %s\n", err)
+	}
+	kernel32 := windows.NewLazySystemDLL("kernel32.dll")
```

Comment:
I can't find anything to say if this is cheep or expensive? Is it ok to recall "NewLazySystemDLL" each time this method is called or should we be caching that?

Reply:
Yeah, I can't find anything clear about this either, but in any case we can make the dll a global since it's lazy loaded anyway.